### PR TITLE
[hud] add page for flaky jobs

### PR DESCRIPTION
**Pull request by clee2000**
Add page for flaky jobs and sql queries.  Takes any job that has been rerun and went from red -> green and any joy that has been annotated as not broken trunk

image

### CODE REVIEW

File: torchci/rockset/commons/__sql/annotated_flaky_jobs.sql [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    job.torchci_classification.line as failureLine,
+    job.torchci_classification.captures as failureCaptures,
+    job.torchci_classification.line_num as failureLineNumber,
+from
+    commons.job_annotation a
+    join commons.workflow_job job on job.id = a.jobID
+    join commons.workflow_run w on w.id = job.run_id
+    and w.head_repository.full_name = a.repo and a.repo = :repo
+where
+    a.annotation != 'BROKEN_TRUNK'
```
Comment (ZainRizvi) [inline torchci/rockset/commons/__sql/annotated_flaky_jobs.sql:26]: why this exception?

**Review by ZainRizvi — APPROVED** [review]
Looking forward to giving this a spin :)

File: torchci/rockset/commons/__sql/annotated_flaky_jobs.sql (hunk 1 above)
Comment (huydhn) [reply torchci/rockset/commons/__sql/annotated_flaky_jobs.sql:26]: I guess the reason is that `BROKEN_TRUNK` is not considered flaky

File: torchci/rockset/commons/__sql/flaky_workflows_jobs.sql [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    workflow_job job
+    inner join workflow_run w on w.id = job.run_id
+    inner join repeats on repeats.run_id = job.run_id and repeats.name = job.name
+where
+    w.repository.full_name = :repo
+    and job._event_time >= PARSE_DATETIME_ISO8601(:startTime)
+    and job._event_time < PARSE_DATETIME_ISO8601(:stopTime)
+    and job.run_attempt = 1
+    and w.head_branch = :branch
+    and (
```
Comment (huydhn) [inline torchci/rockset/commons/__sql/flaky_workflows_jobs.sql:42]: Just curious, does `job.conclusion != 'success' and job.conclusion != 'null'` work here or is there other values to consider?  It doesn't make the logic simpler or shorter much I guess, so this is fine as long as this is consistent across all query lambdas querying flaky jobs

Comment (clee2000) [reply torchci/rockset/commons/__sql/flaky_workflows_jobs.sql:42]: According to https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#workflow_job, theres also neutral, action required, and stale.  I think skipped might also be one but I'm not entirely sure

File: torchci/rockset/commons/__sql/flaky_workflows_jobs.sql [hunk 3]
```diff
@@ -0,0 +1,46 @@
+-- Find workflows that succeeded once they were retried
+with repeats as (
+    select
+        j.run_id,
+        j.name
+    from
+        workflow_job j
+    where
+        j._event_time > CURRENT_TIMESTAMP() - DAYS(7)
```
[the remark this answers is not in the document]
Comment (clee2000) [reply torchci/rockset/commons/__sql/flaky_workflows_jobs.sql]: oops, fixed

**Review by huydhn — APPROVED** [review]

File: torchci/rockset/commons/__sql/flaky_workflows_jobs.sql [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        and j.run_attempt > 1
+        and j.conclusion = 'success'
+) -- When a first time contributor submits a PR, their workflow starts at "attempt 2" so we look for workflows that were repeated above and actually have a first attempt
```
[the remark this answers is not in the document]
Comment (ZainRizvi) [reply torchci/rockset/commons/__sql/flaky_workflows_jobs.sql:13]: Yup, "attempt 1" is the not-even-tried attempt when they make their Pull Request.  It's weird that github categoriezes it this way but 🤷

**Pull request merged**